server/eventlog/eventlog.go

```diff
@@ -23,41 +36,138 @@ func GetEventLogChunk(ctx context.Context, env environment.Env, req *elpb.GetEve
 		return nil, err
 	}
 
+	invocationInProgress := inv.InvocationStatus == int64(inpb.Invocation_PARTIAL_INVOCATION_STATUS)
 	c := chunkstore.New(env.GetBlobstore(), &chunkstore.ChunkstoreOptions{})
-	var intChunkId uint16
-	if len(req.ChunkId) > 0 {
-		n, err := strconv.ParseUint(req.ChunkId, 16, 16)
-		if err != nil {
+	eventLogPath := getEventLogPathFromInvocationId(req.InvocationId)
+
+	lastChunkId, err := c.GetLastChunkId(ctx, eventLogPath, inv.LastChunkId)
+	if err != nil {
+		if inv.LastChunkId != chunkstore.ChunkIndexAsStringId(math.MaxUint16) {
```

Comment:
Hmm I'm not quite following -- what is the effect of the "FFFF" check here? Also, would it make sense to catch `inv.LastChunkId == "FFFF"` earlier in the func and immediately return an err?

Reply:
maybe worth noting: GetLastChunkId correctly interprets "FFFF" as meaning that we should start searching at 0.

Reply:
Would it be equivalent to write `if !status.IsNotFoundError(err)`? To me, that feels like it more directly expresses the intent.

Reply:
Unfortunately, it wouldn't. It is is possible that the status could be `NotFound` with a non-`FFFF` chunk id if, for example, the logs were deleted. The only time it is not an error case for `GetLastChunkId` to fail is specifically if the chunk recorded in the invocation table is `FFFF`. Thus, the status of `err` cannot determine if we are in actual error state.

Notably, it is not guaranteed to fail for `FFFF`, as it will correctly find and return the ID of the last chunk written, if any exist, which is why we only test for this equivalency in the failure case.